src/settings.c

```diff
@@ -289,27 +268,41 @@ static void process_conf_file(const gpointer conf_fname, gpointer n_success) {
         ++(*(int *) n_success);
 }
 
-void load_settings(const char *const path) {
-        // NOTE: settings_init should be called before if some settings are changed
-        //       But having it here is useful for tests
-        //       Potentially, we could update the settings code and move this somewhere else
+void load_settings(char **const config_paths)
+{
+        // NOTE: settings_init should be called at the start of dunst_main, otherwise
+        //       the cmdline settings would be reset
         settings_init();
+
         LOG_D("Setting defaults");
         set_defaults();
 
-        GPtrArray *conf_files = get_conf_files(path);
+        guint length = g_strv_length(config_paths);
+
+        GPtrArray *conf_files;
+
+        if (length != 0) {
+                conf_files = g_ptr_array_new_full(length, g_free);
+                for (int i = 0; config_paths[i]; i++)
+                        g_ptr_array_add(conf_files, g_strdup(config_paths[i]));
+        } else {
+                // Use default locations (and search drop-ins)
+                conf_files = get_conf_files();
+        }
 
         /* Load all conf files and drop-ins, least important first. */
         int n_loaded_confs = 0;
         g_ptr_array_foreach(conf_files, process_conf_file, &n_loaded_confs);
 
         if (0 == n_loaded_confs)
-                LOG_I("No configuration file found, using defaults");
+                LOG_M("No configuration file found, using defaults");
 
-        for (GSList *iter = rules; iter; iter = iter->next) {
-                struct rule *r = iter->data;
-                print_rule(r);
-        }
+        // Maybe add an option to display the rules
+        //for (GSList *iter = rules; iter; iter = iter->next) {
```

Comment:
This code here should be deleted as the referenced function doesn't exist anymore and thus it will be more confusing than helping.

Reply:
Yes, I left it only because we have no way to print rules currently. maybe we should add an option to do it?